node/processingOnlyNode/coreComponents.go

```diff
@@ -0,0 +1,381 @@
+		return nil, err
+	}
+	instance.hasher, err = hashingFactory.NewHasher(args.Cfg.Hasher.Type)
+	if err != nil {
+		return nil, err
+	}
+	instance.txSignHasher, err = hashingFactory.NewHasher(args.Cfg.TxSignHasher.Type)
+	if err != nil {
+		return nil, err
+	}
+	instance.uint64SliceConverter = uint64ByteSlice.NewBigEndianConverter()
+	instance.addressPubKeyConverter, err = factoryPubKey.NewPubkeyConverter(args.Cfg.AddressPubkeyConverter)
+	if err != nil {
+		return nil, err
+	}
+	instance.validatorPubKeyConverter, err = factoryPubKey.NewPubkeyConverter(args.Cfg.ValidatorPubkeyConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	instance.pathHandler, err = storageFactory.CreatePathManager(
+		storageFactory.ArgCreatePathManager{
+			WorkingDir: args.WorkingDir,
+			ChainID:    args.Cfg.GeneralSettings.ChainID,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO check if we need the real watchdog
+	instance.watchdog = &watchdog.DisabledWatchdog{}
+	// TODO check if we need the real alarm scheduler
+	instance.alarmScheduler = alarm.NewAlarmScheduler()
+	// TODO check if we need the real sync time also this component need to be started
+	instance.syncTimer = ntp.NewSyncTime(args.Cfg.NTPConfig, nil)
+	// TODO discuss with Iulian about the round handler
+	//instance.roundHandler
+
+	instance.wasmVMChangeLocker = &sync.RWMutex{}
+	instance.txVersionChecker = versioning.NewTxVersionChecker(args.Cfg.GeneralSettings.MinTransactionVersion)
+	instance.epochNotifier = forking.NewGenericEpochNotifier()
+	instance.enableEpochsHandler, err = enablers.NewEnableEpochsHandler(args.EnableEpochsConfig, instance.epochNotifier)
+	if err != nil {
+		return nil, err
+	}
+
+	argsGasSchedule := forking.ArgsNewGasScheduleNotifier{
+		GasScheduleConfig: config.GasScheduleConfig{
+			GasScheduleByEpochs: []config.GasScheduleByEpochs{
+				{
+					StartEpoch: 0,
+					FileName:   args.GasScheduleFilename,
+				},
+			},
+		},
+		ConfigDir:          "",
+		EpochNotifier:      instance.epochNotifier,
+		WasmVMChangeLocker: instance.wasmVMChangeLocker,
+	}
+	gasScheduleNotifier, err := forking.NewGasScheduleNotifier(argsGasSchedule)
+	if err != nil {
+		return nil, err
+	}
+
+	builtInCostHandler, err := economics.NewBuiltInFunctionsCost(&economics.ArgsBuiltInFunctionCost{
+		ArgsParser:  smartContract.NewArgumentParser(),
+		GasSchedule: gasScheduleNotifier,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	argsEconomicsHandler := economics.ArgsNewEconomicsData{
+		TxVersionChecker:            instance.txVersionChecker,
+		BuiltInFunctionsCostHandler: builtInCostHandler,
+		Economics:                   &args.EconomicsConfig,
+		EpochNotifier:               instance.epochNotifier,
+		EnableEpochsHandler:         instance.enableEpochsHandler,
+	}
+
+	instance.economicsData, err = economics.NewEconomicsData(argsEconomicsHandler)
+	if err != nil {
+		return nil, err
+	}
+	instance.apiEconomicsData = instance.economicsData
+
+	// TODO check if we need this
```

Comment:
I guess we will need this for the system VM. The TODO can be left, we will decide later.